src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,446 @@
+            {
+                return runtimeInformationType.GetMembers().OfType<IMethodSymbol>().Where(m =>
+                    s_platformCheckMethodNames.Contains(m.Name) && !m.Parameters.IsEmpty).ToImmutableArray();
+            }
+        }
+
+        private void AnalyzerOperationBlock(OperationBlockStartAnalysisContext context, ImmutableArray<IMethodSymbol> guardMethods, INamedTypeSymbol osPlatformType)
+        {
+            var platformSpecificOperations = PooledDictionary<IOperation, ImmutableArray<PlatformAttributeInfo>>.GetInstance();
+
+            context.RegisterOperationAction(context =>
+            {
+                AnalyzeInvocationOperation(context.Operation, context, ref platformSpecificOperations);
+            }
+            , OperationKind.DelegateCreation, OperationKind.EventReference, OperationKind.FieldReference,
+            OperationKind.Invocation, OperationKind.ObjectCreation, OperationKind.PropertyReference);
+
+            context.RegisterOperationBlockEndAction(context =>
+            {
+                try
+                {
+                    if (platformSpecificOperations.Count == 0)
+                    {
+                        return;
+                    }
+
+                    if (guardMethods.IsEmpty || !(context.OperationBlocks.GetControlFlowGraph() is { } cfg))
+                    {
+                        ReportDiagnosticsForAll(platformSpecificOperations, context);
+                        return;
+                    }
+
+                    var wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(context.Compilation);
+                    var analysisResult = GlobalFlowStateAnalysis.TryGetOrComputeResult(
+                        cfg, context.OwningSymbol, CreateOperationVisitor, wellKnownTypeProvider,
+                        context.Options, MinimumOsRule, performValueContentAnalysis: true,
+                        context.CancellationToken, out var valueContentAnalysisResult);
+
+                    if (analysisResult == null)
+                    {
+                        ReportDiagnosticsForAll(platformSpecificOperations, context);
+                        return;
+                    }
+
+                    foreach (var platformSpecificOperation in platformSpecificOperations)
+                    {
+                        var value = analysisResult[platformSpecificOperation.Key.Kind, platformSpecificOperation.Key.Syntax];
+                        PlatformAttributeInfo attribute = platformSpecificOperation.Value.FirstOrDefault();
+
+                        if (value.Kind == GlobalFlowStateAnalysisValueSetKind.Unknown)
+                        {
+                            if (platformSpecificOperation.Key.TryGetContainingLocalOrLambdaFunctionSymbol(out var containingSymbol))
+                            {
+                                var localResult = analysisResult.TryGetInterproceduralResultByDefinition(containingSymbol);
```

Comment:
I would suggest looping through all results instead of just the first.

Reply:
I will check

Reply:
As mentioned above, the results return from the single value was enough (we only need the original definition of local function which will be same for all invocation)

Reply:
As you mentioned at other place, this really needs to handle all results, not just the first. I think the fix here would be just to convert `TryGetInterproceduralResultByDefinition` to return an `IEnumerable` and then have the below logic in a loop. Does that not fix the issue?

Reply:
Note that you will need to make a slight change to the below logic - 

```csharp
var hasKnownUnguardedValue = false;
foreach (var localResult in analysisResult.TryGetInterproceduralResultsByDefinition(containingSymbol))
{
    var localValue = localResult[platformSpecificOperation.Key.Kind, platformSpecificOperation.Key.Syntax];
    if (localValue.Kind == GlobalFlowStateAnalysisValueSetKind.Known && !IsKnownValueGuarded(attribute, localValue))
    {
        hasKnownUnguardedValue = true;
        break;
    }
}

if (!hasKnownUnguardedValue)
{
    continue;
}
```

Reply:
> As you mentioned at other place, this really needs to handle all results, not just the first. I think the fix here would be just to convert TryGetInterproceduralResultByDefinition to return an IEnumerable and then have the below logic in a loop. Does that not fix the issue?

That is just the same, even bringing all invocations as an IEnumerable here and loop through the results will be the same because both invocation has the same local method and it will return the value just whichever listed first. Anyway, i have updated it as you wanted, but that would not solve the real problem, here is the test illustrates that:
```csharp
using System.Runtime.Versioning;
using System;
public class Test
{
    void M()
    {
        if (OperatingSystemHelper.IsOSPlatformVersionAtLeast(""Linux"", 10, 2))
        {
            LocalM(true);
        }
        LocalM(false);
        return;

        void LocalM(bool suppressed)
        {
            if (suppressed)
            {
                WindowsOnlyMethod();
            }
            else
            {
                [|WindowsOnlyMethod()|];
            }
            
            if (OperatingSystemHelper.IsOSPlatformVersionAtLeast(""Windows"", 10, 2))
            {
                WindowsOnlyMethod();
            }

            if (OperatingSystemHelper.IsWindows() && !OperatingSystemHelper.IsWindowsVersionAtLeast(10,0))
            {
                UnsupportedWindows10();
            }
            else
            {
                [|UnsupportedWindows10()|];
            }
        }
    }
    [SupportedOSPlatform(""Windows10.1.2.3"")]
    public void WindowsOnlyMethod()
    {
    }
    [UnsupportedOSPlatform(""Windows10.0"")]
    public void UnsupportedWindows10()
    {
    }
}
```